### ISSUE TITLE
Fix "deprecated pixel format used" error message

### ISSUE THREAD

**Issue by juren53** [juren53/HST-Metadata#12]
ffmpeg error message:

[swscaler @ 0x56396bbdd000] deprecated pixel format used, make sure you did set range correctly

This message from FFmpeg indicates that the software has detected the use of a deprecated pixel format in the video processing. Here's a breakdown of the message:

    [swscaler @ 0x7f39ec05a5c0]: This part of the message specifies the module (swscaler, which is the software scaler library used for image scaling and format conversion) and the memory address where the issue is occurring.

    deprecated pixel format used: FFmpeg is informing you that the pixel format being used is deprecated, meaning it is outdated and might be removed in future versions.

    make sure you did set range correctly: This suggests that you should check the range settings for your pixel format. Pixel formats can have different range settings (e.g., limited range or full range), and using the correct one is important for proper video processing.

To resolve this issue, you may need to:

    Identify the Deprecated Pixel Format: Determine which pixel format you are using that is deprecated. This can be done by checking the FFmpeg command or script you are using.

    Update to a Supported Pixel Format: Replace the deprecated pixel format with a supported one. You can find a list of supported pixel formats in the [FFmpeg documentation](https://ffmpeg.org/ffmpeg-filters.html#pixel-formats).

    Set the Range Correctly: Ensure that you are using the correct range setting for your pixel format. This can usually be specified in your FFmpeg command with options like -pix_fmt for pixel format and -color_range for range settings.

Here's an example command to convert a video with a specific pixel format and range setting:

  bash

  ffmpeg -i input.mp4 -pix_fmt yuv420p -color_range tv output.mp4

In this example, yuv420p is a commonly used pixel format, and tv sets the color range to limited 

(which is typical for broadcast TV).

Updating your FFmpeg commands and scripts to use supported pixel formats and correct range 

settings should resolve the warning message.

**Comment by juren53** [comment]
juren@jaus-e6410-141:~/Projects/HST-Metadata/Audio-Metadata/Test/processed$ tgpt "what does this ffmpeg message mean:
> [swscaler @ 0x7f39ec05a5c0] deprecated pixel format used, make sure you did set range correctly
"
         
The message you're seeing from FFmpeg indicates that there's an issue related to the pixel format being used during the processing of video
 frames. Specifically, it mentions:

``
[swscaler @ 0x7f39ec05a5c0] deprecated pixel format used, make sure you did set range correctly
```

This message suggests two main points:

1. **Deprecated Pixel Format**: The pixel format being used is outdated or no longer recommended. This could potentially lead to issues with
 compatibility or performance.

2. **Set Range Correctly**: It advises ensuring that the color range (also known as the color space) is correctly configured. In video
 processing, color range refers to how colors are represented in terms of their intensity values. There are typically two ranges: full (or "
limited") range, where all possible color values are used, and limited range, which restricts certain high-intensity values to prevent
 clipping when displayed on standard TV sets.

To resolve this issue, you might need to explicitly set the correct color range for your input or output streams. Here's how you can do it
 using FFmpeg commands:

- To set the color range for an input stream, use the -vf option followed by format=yuv420p (for example) to ensure it uses a specific
 pixel format that supports the desired color range. For instance:

```bash
ffmpeg -i nput.mp4 -vf format=yuv420p outut.mp4
```

- If you're dealing with an output stream and want to ensure it uses a specific color range, you can use the -pix_fmt option. For example,
 to set the output to use the YUV color space with a limited range:

```bash
ffmpeg -i nput.mp4 -pix_fmt yuv420p outut.mp4
```

Adjusting the pixel format (yuv420p, yuv422p, etc.) and setting the correct color range should help address the deprecation warning and
 ensure your video processing tasks run smoothly.

juren@jaus-e6410-141:~/Projects/HST-Metadata/Audio-Metadata/Test/processed$

**Comment by juren53** [comment]
juren@jaus-e6410-141:~/Projects/HST-Metadata/Audio-Metadata/Test$ tgpt "what does this ffmpeg message mean:
> [Parsed_drawtext_0 @ 0x559dbd367ac0] Using "/usr/share/fonts/truetype/ttf-bitstream-vera/Vera.ttf"
> "
          
The message you're seeing from FFmpeg is related to the drawtext filter, which is used to overlay text onto video frames. The specific part  of the message [Parsed_drawtext_0 @ 0x559dbd367ac0] Using /usr/share/fonts/truetype/ttf-bitstream-vera/Vera.ttf indicates that FFmpeg is  using the Vera.ttf font located at /usr/share/fonts/truetype/ttf-bitstream-vera/ to render the text.

The rest of the message, "\\n\\nHere is text for the context:\\n", shows the actual text content that will be drawn on the video frame(s).  In this case, it appears there might be an issue with how the newline characters (\n) are being interpreted or displayed. The double back slashes (\\) before each \n suggest that the text string is being escaped, meaning each \n is intended to represent a newline character  in the final output. However, the way it's displayed in the log message makes it seem like the escape sequences aren't being processed  correctly, leading to the literal display of \n instead of a newline.

This could be due to how the logging system interprets escape sequences or simply a formatting issue in the log output. If your goal is to  have the text appear on separate lines in the video output, ensure that the text string passed to the drawtext filter includes the correct  number of newline characters without them being misinterpreted in the log messages.

**Comment by juren53** [comment]
tgpt "if I don't specify a font when I access the drawtext filter in FFmpeg, where does the default font come from?"
          
In FFmpeg, when you use the drawtext filter without specifying a font, it defaults to using the system's default font. This means that the appearance of the text will depend on the fonts installed on the machine where FFmpeg is being executed. The specific font used can vary significantly across different systems due to the diversity of installed fonts.

If you want to ensure consistent results across different environments, it's recommended to explicitly specify a font using the -fontfile option followed by the path to the desired font file. For example:

```bash
ffmpeg -i nput.mp4 -vf "drawtext=='Hello World':fontfile=/path/to/your/font.ttf" output.mp4
```

This approach allows you to control exactly which font is used, making your video processing more predictable and portable across different  systems.

**Comment by juren53** [comment]
juren@jaus-e6410-141:~/Projects/HST-Metadata/Audio-Metadata/Test$ tgpt -provider blackboxai "how did /usr/share/fonts/truetype/ttf-bitstream-vera/Vera.ttf get selected as the font when I ran an FFmpeg command when I did not specify the font?"
          

$@$v=v1.20-rv1$@$When you run an FFmpeg command that involves text rendering, such as adding subtitles or text overlays, FFmpeg needs to choose a font to use. If you don't specify a font explicitly, FFmpeg will fall back to a default font.



The default font selection process is not well-documented, but based on the FFmpeg source code and various online resources, here's what happens:


1. **FFmpeg checks the environment variable FONTCONFIG_FILE**: If this variable is set, FFmpeg will use the font configuration file specified in the variable to determine the default font.

2. **FFmpeg checks the system's font configuration**: If FONTCONFIG_FILE is not set, FFmpeg will use the system's font configuration, which is usually managed by the fontconfig library. Fontconfig is a library that provides a common interface for font management on Linux and other Unix-like systems.

3. **Fontconfig searches for a suitable font**: Fontconfig will search for a font that matches the language and script of the text being rendered. In your case, since you didn't specify a language or script, Fontconfig will likely default to a font that supports the Latin script (e.g., English).

4. **Fontconfig returns a default font**: Fontconfig will return a default font that matches the search criteria. In your case, it seems that Fontconfig returned the Vera.ttf font from the /usr/share/fonts/truetype/ttf-bitstream-vera/ directory.


The Vera.ttf font is a part of the Bitstream Vera font family, which is a popular open-source font family designed for use on the web. It's commonly included in many Linux distributions, which is why it's available on your system.

So, to summarize, the Vera.ttf font was selected as the default font because:
* You didn't specify a font explicitly in your FFmpeg command.* The FONTCONFIG_FILE environment variable was not set.
* Fontconfig, the system's font management library, searched for a suitable font and returned Vera.ttf as the default font.

If you want to use a different font, you can specify it explicitly in your FFmpeg command using the -font option, like this:bash
ffmpeg -i input.mp4 -vf drawtext=text='Hello World':fontfile=/path/to/your/font.ttf output.mp4
```
Replace /path/to/your/font.ttf with the path to the font file you want to use.


juren@jaus-e6410-141:~/Projects/HST-Metadata/Audio-Metadata/Test$
